src/pdb/mod.rs

```diff
@@ -1184,4 +1304,634 @@ mod test {
+                        orig_artist_id: 0,
+                        label_id: 0,
+                        remixer_id: 0,
+                        bitrate: 2116,
+                        track_number: 0,
+                        tempo: 0,
+                        genre_id: 0,
+                        album_id: 0,
+                        artist_id: 0,
+                        id: 1,
+                        disc_number: 0,
+                        play_count: 0,
+                        year: 0,
+                        sample_depth: 24,
+                        duration: 5,
+                        unknown5: 41,
+                        color: ColorIndex::None,
+                        rating: 0,
+                        unknown6: 11,
+                        unknown7: 3,
+                        isrc: DeviceSQLString::new_isrc("".to_string()).unwrap(),
+                        unknown_string1: DeviceSQLString::empty(),
+                        unknown_string2: DeviceSQLString::new("3".to_string()).unwrap(),
+                        unknown_string3: DeviceSQLString::new("2".to_string()).unwrap(),
+                        unknown_string4: DeviceSQLString::empty(),
+                        message: DeviceSQLString::empty(),
+                        kuvo_public: DeviceSQLString::empty(),
+                        autoload_hotcues: DeviceSQLString::new("ON".to_string()).unwrap(),
+                        unknown_string5: DeviceSQLString::empty(),
+                        unknown_string6: DeviceSQLString::empty(),
+                        date_added: DeviceSQLString::new("2015-09-07".to_string()).unwrap(),
+                        release_date: DeviceSQLString::empty(),
+                        mix_name: DeviceSQLString::empty(),
+                        unknown_string7: DeviceSQLString::empty(),
+                        analyze_path: DeviceSQLString::new(
+                            "/PIONEER/USBANLZ/P019/00020AA9/ANLZ0000.DAT".to_string(),
+                        )
+                        .unwrap(),
+                        analyze_date: DeviceSQLString::new("2022-02-02".to_string()).unwrap(),
+                        comment: DeviceSQLString::empty(),
+                        title: DeviceSQLString::new("NOISE".to_string()).unwrap(),
+                        unknown_string8: DeviceSQLString::empty(),
+                        filename: DeviceSQLString::new("NOISE.wav".to_string()).unwrap(),
+                        file_path: DeviceSQLString::new(
+                            "/Contents/UnknownArtist/UnknownAlbum/NOISE.wav".to_string(),
+                        )
+                        .unwrap(),
+                    }),
+                ],
+                row_presence_flags: 96,
+                unknown: 64,
+            }],
+        };
+
+        let page_size: u32 = 4096;
+        test_roundtrip_with_args(
+            &[
+                0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 47, 0, 0, 0, 32, 0, 0, 0, 0, 0, 0, 0, 7, 64, 0,
```

Comment:
Where did you get this from? Just exporting from rekordbox?

Reply:
Yes, IIRC It's a page from the rekordbox export in this repo. I should've documented this in a comment.

Reply:
I'll add a comment when I open the PR with the solution.